### PR TITLE
Botiana's Big Update

### PR DESCRIPTION
**Pull request by rpkish**
Botiana has been made a little more sane, and upgraded for python 3.7

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,29 +1,20 @@
...
-    python-yaml \
- && rm -rf /var/lib/apt/lists/* \
- && pip install BeautifulSoup \
- && pip install beautifulsoup4 
+# Build deps
+RUN apk update \
+    && apk add --virtual build-dependencies build-base py3-lxml libxml2-dev libxslt-dev
 
 # Install botiana and copy modified settings and dictionaries 
 RUN mkdir /usr/local/botiana/
```
Comment (diversario) [inline Dockerfile]: iirc `COPY` will make one for you implicitly

File: Dockerfile [hunk 2]
```diff
@@ -1,29 +1,20 @@
...
-COPY ./botiana.py ./settings.py ./sa.yaml ./keywords.yaml ./requirements.txt /usr/local/botiana/
+COPY ./botiana.py ./settings.py ./requirements.txt ./common.py ./keywords.py ./legacy_modules.py  \
+     ./message_router.py ./settings.py ./slack_commands.py /usr/local/botiana/
 
 # run configuration steps
 RUN pip install -r /usr/local/botiana/requirements.txt
 
 # Fire up botiana
-ENTRYPOINT cd /usr/local/botiana && ./botiana.py
+ENTRYPOINT cd /usr/local/botiana && /usr/local/bin/python3 -u ./botiana.py
```
Comment (diversario) [inline Dockerfile]: put `WORKDIR /usr/local/botiana` before `ENTRYPOINT`
Also, do you have to use `ENTRYPOINT` over `CMD`?

File: keywords.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+from random import randint
+import random
+from slack_commands import __send_response, __reaction
+
+
+def keywords(sc, evt, yamldata, count):
+    if "channel" in evt and evt["type"] == "message":
+        message = evt["text"].strip()
+        if "bot_id" not in evt:
```
Comment (diversario) [inline keywords.py]: If you're not opposed to this style, short-circuiting
```
if "bot_id" in evt:
  return count

<rest of the stuff
```
would reduce indentation level.

File: keywords.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            for k in yamldata["keywords"]:
+                if k in message.lower():
+                    # Intent here is not to trigger every time.
+                    # Increment on times seen, trigger a random recount vault
+                    # when we deploy a message
+                    if count["current"] > count["random"]:
+                        # Does our entry only apply to a specific channel?
+                        if yamldata["keywords"][k]["watch"] != "all":
+                            # Does our entry apply to all channels?
```
Comment (diversario) [inline keywords.py]: wrong comment?

File: keywords.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        message = evt["text"].strip()
+        if "bot_id" not in evt:
+            for k in yamldata["keywords"]:
+                if k in message.lower():
+                    # Intent here is not to trigger every time.
+                    # Increment on times seen, trigger a random recount vault
+                    # when we deploy a message
+                    if count["current"] > count["random"]:
+                        # Does our entry only apply to a specific channel?
+                        if yamldata["keywords"][k]["watch"] != "all":
```
Comment (diversario) [inline keywords.py]: Maybe
```
if yamldata["keywords"][k]["watch"] != "all" and yamldata["keywords"][k]["watch"] == evt["channel"]
```
?

Or, is there def a need to do this `all` check, vs just
```
if yamldata["keywords"][k]["watch"] == evt["channel"]:
```
?

Comment (diversario) [reply keywords.py]: It looks like the content of `if/else` blocks is duplicated. Changing the check probably can help reduce it to one `if` block.

File: Dockerfile (hunk 1 above)
Comment (rpkish) [reply Dockerfile]: Dockerfile updated. Thanks!

File: Dockerfile (hunk 2 above)
Comment (rpkish) [reply Dockerfile]: Dockerfile updated, thanks!

File: keywords.py (hunk 3 above)
Comment (rpkish) [reply keywords.py]: this was a great suggestion, it removed a ton of redundant code too. Thanks!

**Pull request merged**